src/hooks/useWarning.js

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect } from 'react';
```

Comment:
This whole file uses 4 spaces for indentation instead of the 2 spaces used by every other file. Please fix. Setup eslint if you want to see and fix these sorts of erros.